### PR TITLE
[HDRP] Fix tube light mode

### PR DESCRIPTION
**Pull request by alelievr**
### Purpose of this PR
Fixed https://fogbugz.unity3d.com/f/cases/resolve/1317776/

---
### Testing status
Played a bit with area light type enum:
Light21

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/CHANGELOG.md [hunk 1]
```diff
@@ -94,6 +94,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Fixed gizmo rendering when wireframe mode is selected.
 - Fixed issue in path tracing, where objects would cast shadows even if not present in the path traced layers (case 1318857).
 - Fixed SRP batcher not compatible with Decal (case 1311586)
+- Fixed Tube Light mode selection (case 1317776)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/CHANGELOG.md:97]: Fix tube and disc light

**Review by sebastienlagarde — APPROVED** [review]

File: com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.cs [hunk 2]
```diff
@@ -208,8 +208,14 @@ static void DrawGeneralContent(SerializedHDLight serialized, Editor owner)
...
-            //Draw the mode
-            serialized.settings.DrawLightmapping();
+            // Draw the mode, for Tube and Disc lights, there is only one choice, so we can disable the enum.
+            if (serialized.areaLightShape == AreaLightShape.Tube || serialized.areaLightShape == AreaLightShape.Disc)
+            {
+                using (new EditorGUI.DisabledScope(true))
+                    serialized.settings.DrawLightmapping();
+            }
+            else
+                serialized.settings.DrawLightmapping();
```
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.cs:218]: why not
```
using (new EditorGUI.DisabledScope(serialized.areaLightShape == AreaLightShape.Tube || serialized.areaLightShape == AreaLightShape.Disc))
      serialized.settings.DrawLightmapping();
```

**Review by RSlysz — APPROVED** [review]

File: com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.cs (hunk 2 above)
Comment (alelievr) [reply com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.cs]: Yes

**Comment by sebastienlagarde** [comment]
QA: simple enough so I am merging

**Pull request merged**